virtual_rainforest/models/soil/carbon.py

```diff
@@ -116,20 +137,28 @@ def mineral_association(
 
         # Calculate
         Q_max = calculate_max_sorption_capacity(bulk_density, percent_clay)
-        equib_maom = calculate_equilibrium_maom(pH, Q_max, self.lmwc)
+        equib_maom = calculate_equilibrium_maom(
+            pH, Q_max, data["low_molecular_weight_c"]
+        )
 
         # Find scalar factors that multiple rates
         temp_scalar = convert_temperature_to_scalar(soil_temp)
         moist_scalar = convert_moisture_to_scalar(soil_moisture)
 
-        return temp_scalar * moist_scalar * self.lmwc * (equib_maom - self.maom) / Q_max
+        return (
+            temp_scalar
+            * moist_scalar
+            * data["low_molecular_weight_c"]
+            * (equib_maom - data["mineral_associated_om"])
+            / Q_max
```

Comment:
It seems a little odd to have each variable passed in as a dataarray , _except_ for `maom` and `lwmc`, which are wrapped up within `data` still? Is there any reason not to pass those arrays in?

Reply:
This is basically an artefact of `maom` and `lmwc` previously being attributes of `SoilCarbonPools`. When they stopped being attributes I changed how they were accessed without making them consistent with what had gone before.

However, I am tempted to leave them as is, as the way this data is accessed has already been changed on my integration branch. So, it might be more productive to have the DataArray vs NDArray discussion when the pull request for that is in?